pipeline/coordinator.go

```diff
@@ -188,6 +199,46 @@ func NewStubCoordinatorOpts(strategy Strategy, statusClient clients.TranscodeSta
 // This has the main logic regarding the pipeline strategy. It starts jobs and
 // handles processing the response and triggering a fallback if appropriate.
 func (c *Coordinator) StartUploadJob(p UploadJobPayload) {
+	// A bit hacky - this is effectively a dummy job object to allow us to reuse the runHandlerAsync and
+	// progress reporting logic. The real job objects still get created in startOneUploadJob().
+	si := &JobInfo{
+		UploadJobPayload: p,
+		statusClient:     c.statusClient,
+		startTime:        time.Now(),
+
+		numProfiles:    len(p.Profiles),
+		state:          "segmenting",
+		catalystRegion: os.Getenv("MY_REGION"),
+	}
+	si.ReportProgress(clients.TranscodeStatusPreparing, 0)
+
+	c.runHandlerAsync(si, func() (*HandlerOutput, error) {
```

Comment:
Why are we using runHandlerAsync here rather than a standard `go func(){}`?

Reply:
So that we can re-use the logic to report errors to studio via the callback handler.

Reply:
Ah yep, gotcha

Reply:
@victorges would you mind sanity checking my usage of this function as I believe you wrote this stuff originally? It's tested in staging successful and failed ingests and I can see the callbacks are working but are there any edge cases you think I should check?